### PR TITLE
Fail ignore sink matval on abrupt termination 

### LINKED ISSUE #23726: IgnoreSink doesn't complete mat future on abrupt termination
Needs to always fail promise `onStop` if not already completed

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #23726

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala [hunk 1]
```diff
@@ -433,7 +433,12 @@ import scala.concurrent.{ Future, Promise }
           promise.tryFailure(ex)
         }
 
+        override def postStop(): Unit = {
+          if (!promise.isCompleted) promise.tryFailure(new AbruptStageTerminationException(this))
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: the if is not needed I'd say

**Review by ktoso — APPROVED** [review]
LGTM

File: akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala (hunk 1 above)
Comment (johanandren) [reply akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: Just wanted to avoid creating the exception in the normal case.

Comment (ktoso) [reply akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala]: Yeah good reason actually, LGTM :)

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**